### PR TITLE
chore: add ancestor runtime ID

### PR DESCRIPTION
**Pull request by P403n1x87**
Because of the current implementation of the remote configuration client, configuration requests are made by the common ancestor process. This means that child processes are invisible to the RCM backend. Some products require information about the runtime ID to provide a good UX. We add the possibility of retrieving the runtime ID of the common ancestor process (the one that will communicate with the RCM backend) from any child so that its runtime ID can be correlated with the ancestor, and used by backend services that rely on the enablement signal coming from RCM requests.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by shatzi — COMMENTED** [review]
nice!
should we also update the probe-status messages?

**Comment by P403n1x87** [comment]
> nice! should we also update the probe-status messages?

Yep, we'll do that in #7912

File: ddtrace/internal/runtime/__init__.py [hunk 1]
```diff
@@ -27,3 +34,13 @@ def get_runtime_id():
     Do not store this identifier as it can change when, e.g., the process forks.
     """
     return _RUNTIME_ID
+
+
+def get_ancestor_runtime_id() -> t.Optional[str]:
+    """Return the runtime ID of the common ancestor of this process.
+
+    Once this values is set (this will happen after a fork) it will not change
```
Comment (ZStriker19) [inline ddtrace/internal/runtime/__init__.py:42]:
```suggestion
    Once this value is set (this will happen after a fork) it will not change
```

**Review by majorgreys — COMMENTED** [review]
Being the changes are internal features, should we title this PR differently?

**Review by majorgreys — CHANGES_REQUESTED** [review]
Change the PR title to a chore to be consistent with the convention we have been using recently for "internal", non-public facing feature work (which this seems to be if I am following).

**Review by majorgreys — APPROVED** [review]

**Pull request merged**